### ISSUE TITLE
which fingerprint to use when multiple fingerprint for one molecule was generate

### ISSUE THREAD

**Issue by kexul** [keiserlab/e3fp#45]
After trying e3fp, I found that , for one molecule, many fingerprint can be generated. Which fingerprint should I use when performing machine learning tasks?

**Comment by sethaxen** [comment]
There are a number of configurable options for e3fp.

e3fp operates on conformers, computing one fingerprint for each conformer, so the first choice is the conformer library for the molecule. I don't know of a clear benefit of one library over another for e3fp's performance. We have included an open source conformer generation method in `e3fp.conformer.generate`, which we used in the paper.

There are also several configurable options for the fingerprinting process. We optimized over these parameters in the paper and selected a well-performing parameter set for our tasks, which are the default parameters used in this package. However, different parameters could perform better for other tasks. I recommend reading the [e3fp paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6075869/) if you have not already done so.

Having selected a conformer library and fingerprinting options, e3fp will generate one fingerprint for each conformer. If your question is which of these to use, we don't really know. Ideally you would use more than one, as a single conformer does not reflect the conformational heterogeneity of the molecule. If you had to use only one, the predicted lowest energy conformer would be a reasonable choice (conformers produced using `e3fp.conformer.generate` are sorted in order of increasing energy). But we also explored in the paper combining bit fingerprints for multiple conformers into a single counts fingerprint or float fingerprint (via averaging). However, these techniques have not been benchmarked as thoroughly.
